16_microbiome_community.Rmd

```diff
@@ -210,6 +210,52 @@ euclidean_dmm_plot <- ggplot(data = euclidean_dmm_pcoa_df,
 euclidean_dmm_plot
 ```
 
+## Community Detection
+
+Another approach for discovering communities within the samples of the data under question,
+is to run community detection algorithms after building a graph. The following demonstration 
+builds a graph based on the k nearest-neighbors and performs the community detection on the fly.
+
+_`bluster`_ [@R-bluster] package offers several clustering methods, among which graph-based are
+present, enabling the community detection task.
+
+Installing package:
+
+```{r}
+if(!require(bluster)){
+  BiocManager::install("bluster")
+}
+```
+
+The algorithm used is "short random walks"[@Pons2006]. 
+Graph is constructed using different k values (the number of nearest neighbors to consider during graph construction) 
+using the clr assay data. Then plotting the communities using the earlier PCoA ordination as a visual exploration aid.
+
+```{r, message=FALSE, warning=FALSE}
+library(bluster)
+library(patchwork) # For arranging several plots as a grid
+
+k <- c(2,3,5,10)
+ClustAndPlot <- function(x) {
+  # Creating the graph and running the short random walks algorithm  
+  graph_clusters <- clusterRows(t(assays(tse)$clr), NNGraphParam(k=x))
+euclidean_pcoa_df$communities <- graph_clusters
+# Results of the clustering as a color for each sample  
+ggplot(data = euclidean_pcoa_df, 
+                             aes(x=pcoa1, y=pcoa2,
+                                 color = communities)) +
+  geom_point() +
+  labs(x = "PC1",
+       y = "PC2",
+       title = paste0("k = ", x))
+}
```

Comment:
Is the indentation little bit weird here? First 2 lines of function are correct, but all other do not have indentation

Reply:
yes it is weird, I fix it.